src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -961,14 +962,20 @@ private void Advance()
             _ilg.Emit(OpCodes.Br, AdvanceLabel());
         }
 
-        private void CallToLower()
+        private void InitLocalCultureInfo()
         {
             if ((_options & RegexOptions.CultureInvariant) != 0)
                 Call(s_getInvariantCulture);
             else
                 Call(s_getCurrentCulture);
 
-            Call(s_chartolowerM);
+            Stloc(_cultureV);
+        }
+
+        private void CallToLower()
+        {
+             Ldloc(_cultureV);
```

Comment:
What is the lifetime of this?  e.g. does this cache end up spanning multiple regex calls and user code such that user code could change the current culture and end up with different behavior than before?

Reply:
_cultureV is a local variable. The lifetime of the cache is therefore method local in FindFirstChar and Go methods of the compiled regular expression.

Reply:
Ah, I see. Thanks.